### ISSUE TITLE
Default styles per-widget should be done per-slot, not absolutely

### ISSUE THREAD

**Issue by noahgibbs** [scarpe-team/scarpe#430]
Have a look at examples/legacy/not_checked/expert/video-player.rb. Initially, it sets the Link and LinkHover styles:

```
Shoes.app width: 625, height: 580, resizable: true do

    LinkStyleStopped = [Shoes::Link, stroke: black, underline: "none"]
    LinkStyleStoppeddHover = [Shoes::LinkHover, stroke: darkred, underline: "none"]
    style(*LinkStyleStopped); style(*LinkStyleStoppeddHover)
```

That's fine once we support LinkHover. However, it also sets up some interesting slot references:

```
        @controls = flow margin: [0,10,0,0], height: 47 do
            # slot needs a height, here, for background to work correctly
            # an alternative is to put @controls slot following (gui) code into a @controls.start {} block
            @bckgrd = background rgb(242,241,240)

            #... (flow widgets continue)
        end
```

And then it sets the default styles for Link and LinkHover *just for that flow*:

```
    def set_controls(color=gray)
        @bckgrd.fill = color
        @controls.style(Shoes::LinkHover, stroke: black, underline: "none")
        @controls.style(Shoes::Link, stroke: lawngreen, underline: "none")
        @ctrls.text = CtrlsText # won't refresh until mouse hovering, otherwise
        @vol_knob.tint = lawngreen
    end
```

So it would appear the for every slot there's a set of default styles per-Shoes-class that must be treated as a hierarchy. That's not what we're doing right now.

**Comment by noahgibbs** [comment]
Link should support stroke and fill styles. The examples/legacy/not_checked/simple/menu.rb example wants them and we don't have them.

**Comment by noahgibbs** [comment]
We should re-do how we handle default values, even from my not-yet-merged PR. I've checked Shoes3 and defaults are applied dynamically at draw-time, not when the drawable is created. And that's part of how they support e.g. LinkHover -- they change the class of the drawable's default-property lookup temporarily, while the user is hovering over it.

**Comment by noahgibbs** [comment]
I explored more and I was wrong about the dynamic handling of default properties. Changing them repeatedly doesn't affect drawables that are already created (though it would with new ones.)

**Comment by noahgibbs** [comment]
I should do more exploring on the per-slot defaults above. I'm not sure if it really works that way or not. Though draw context is definitely done per-slot, and that's implemented in #513.

**Comment by noahgibbs** [comment]
Moved this to Feature Brainstorming. This is pretty deep into obscure features. https://github.com/scarpe-team/scarpe/wiki/DynamicDefaultsPerDrawablePerSlot.md
